sdk/eventhub/Azure.Messaging.EventHubs/src/RetryOptions.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Messaging.EventHubs.Core;
+
+namespace Azure.Messaging.EventHubs
+{
+    /// <summary>
+    ///    The set of options that can be specified to influence how
+    ///    retry attempts are made, and a failure is eligible to be retried.
+    /// </summary>
+    ///
+    public class RetryOptions
+    {
+        /// <summary>The maximum number of retry attempts before considering the associated operation to have failed.</summary>
+        private int _maximumRetries = 3;
+
+        /// <summary>The delay or backoff factor to apply between retry attempts.</summary>
+        private TimeSpan _delay = TimeSpan.FromSeconds(0.8);
+
+        /// <summary>The maximum delay to allow between retry attempts.</summary>
+        private TimeSpan _maximumDelay = TimeSpan.FromMinutes(1);
+
+        /// <summary>The maximum duration to wait for an operation, per attempt.</summary>
+        private TimeSpan _tryTimeOut = TimeSpan.FromMinutes(1);
+
+        /// <summary>
+        ///   The approach to use for calculating retry delays.
+        /// </summary>
+        ///
+        public RetryMode Mode { get; set; } = RetryMode.Exponential;
+
+        /// <summary>
+        ///   The maximum number of retry attempts before considering the associated operation
+        ///   to have failed.
+        /// </summary>
+        ///
+        public int MaximumRetries
```

Comment:
My choice here is to be internally consistent with the Event Hubs naming, which follows the guidelines for "don't abbreviate" and uses maximum or fully mirror the `Core` retry options.   I don't see this as a case where the slight deviation will cause confusion for those using other SDKs with the `Core` options but it would potentially feel wrong to those using Event Hubs.